src/Service/TokenService.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace PrestaShop\Module\PsAccounts\Service;
+
+use PrestaShop\Module\PsAccounts\Adapter\ConfigurationKeys;
+use PrestaShop\Module\PsAccounts\Log\Logger;
+use PrestaShop\Module\PsAccounts\Repository\ConfigurationRepository;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\Clock\FrozenClock;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Builder;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Configuration;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Signer\Hmac\Sha256;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Signer\Key\InMemory;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Token;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Validation\Constraint\SignedWith;
+use PrestaShop\Module\PsAccounts\Vendor\Lcobucci\JWT\Validation\Constraint\ValidAt;
+
+class TokenService
```

Comment:
Find a better name :
```suggestion
class AdminTokenService
```